### DISCUSSION TITLE
WIP - Python Morsels x Black Python Devs

### DISCUSSION THREAD

**Discussion by kjaymiller** [BlackPythonDevs/blackpythondevs#31] [Ideas]
# Project Idea: Python Learning Cohorts

## Time Frame
- Quarterly

### Cohorts
Cohorts are made up of 8 people of similar skill levels to ensure personalized attention and a supportive learning environment. Each cohort is designed to cater to different levels of Python proficiency:
  - **Beginner:** "I know little or no Python." This group is for those starting from scratch, focusing on basics and foundational concepts.
  - **Intermediate:** "I know some Python but likely not using it in my day-to-day." Aimed at those looking to solidify their understanding and start applying Python in more complex scenarios.
  - **Advanced:** "I use Python in my job daily and I'm hoping to continue growing." This cohort is for individuals seeking to learn advanced Python features and best practices.

### Content

Each week the cohorts are encourage to work on the scheduled Python Morsel. Morsels are broken into solutions with bonus solutions criteria. 

The expectation is that beginners would focus on the primary criteria, while intermediate and advanced cohorts would complete some/all of the bonuses.

### Duration

Each cohort will run for a quarter, with weekly modules and assignments.

After the quarter the cohort will receive an opportunity to continue to use Python Morsels at a discounted rate (for the first year). And are invited to stay in the channels where cohorts and alumni are talking about challenges.

### Method

The expectation is that members of the cohort will chat throughout the week on the challenge, helping one another.

Twice monthly, the cohort is expected to meet via the Discord voice channel to share some of their completion strategies with one another.

## Questions
- **Enrollment:** How will participants apply or be selected for a cohort?
- **Progress Tracking:** What mechanisms will be in place to monitor participant progress and engagement?
- **Feedback:** How will feedback be collected and used to improve future cohorts?

## Follow Up
- **Advance to more challenging cohorts:** For those who have completed the beginner cohort, intermediate morsels will offer harder exercises and more depth on topics covered.
- **Networking:** Opportunities for alumni of all cohorts to network, share experiences, and collaborate on projects.

## Evaluation
- **Success Metrics:** The cohort leader is expected to debrief with the leadership team halfway and at the end of the cohort cycle.
- **Alumni Support:** Continued access to learning materials and community support post-completion to encourage lifelong learning and professional growth.

## Conclusion
This project aims to create a structured, supportive, and scalable learning environment for Python enthusiasts at all levels. By fostering a community of learners and providing targeted resources and challenges, we hope to contribute significantly to each participant's professional development.

**Comment by lazouich** [comment]
To echo what I said in the _Coffee and Code_ this morning - I like this idea and would be interested in begin involved. I don't want financing to discourage anyone with partaking. Also, would there be any type of group moderation?
